express-notes-api/index.js

```diff
@@ -0,0 +1,81 @@
+const express = require('express');
+const app = express();
+const fs = require('fs');
+
+app.get('/api/notes', (req, res) => {
+  const data = require('./data.json');
+  res.status(200).json(data.notes);
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  const data = require('./data.json');
+  const id = req.params.id;
+  if (!parseInt(id) || id < 0) {
+    res.status(400).send({ error: 'ID must be a positive integer' });
+  } else if (!data.notes[id]) {
+    res.status(404).send(`cannot find id ${id}`);
+  }
+  res.status(200).json(data.notes[id]);
```

Comment:
Be sure that it is impossible for the server to try to respond more than once to the same request. You'll need an `else` in here.

Reply:
Please add the `else` here too.